ShopAppTests/App/Account/Controllers/AddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,327 @@
+                expect(viewModelMock.firstNameText.value) == "First name"
+            }
+            
+            it("should have correct view model lastNameText binding") {
+                lastNameTextFieldView.text = "Last name"
+                lastNameTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.lastNameText.value) == "Last name"
+            }
+            
+            it("should have correct view model addressText binding") {
+                addressTextFieldView.text = "Address"
+                addressTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.addressText.value) == "Address"
+            }
+            
+            it("should have correct view model addressOptionalText binding") {
+                addressOptionalTextFieldView.text = "Address"
+                addressOptionalTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.addressOptionalText.value) == "Address"
+            }
+            
+            it("should have correct view model cityText binding") {
+                cityTextFieldView.text = "City"
+                cityTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.cityText.value) == "City"
+            }
+            
+            it("should have correct view model stateText binding") {
+                statePicker.text = "State"
+                statePicker.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.stateText.value) == "State"
+            }
+            
+            it("should have correct view model zipText binding") {
+                zipCodeTextFieldView.text = "Zip"
+                zipCodeTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.zipText.value) == "Zip"
+            }
+            
+            it("should have correct view model phoneText binding") {
+                phoneTextFieldView.text = "Phone"
+                phoneTextFieldView.textField.sendActions(for: .editingChanged)
+                expect(viewModelMock.phoneText.value) == "Phone"
+            }
+        }
+        
+        describe("when pickers did changes") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                initData(addressNeeded: false)
+                disposeBag = DisposeBag()
+            }
+            
+            it("should have correct view model namesOfCountries binding") {
+                viewController.viewModel.namesOfCountries
+                    .subscribe(onNext: { names in
+                        expect(names).toEventually(equal(["Country1"]))
+                        expect(countryPicker.customData).toEventually(equal(["Country1"]))
+                    })
+                    .disposed(by: disposeBag)
+                
+                viewModelMock.makeNameOfCountries()
```

Comment:
Make arguments for this method and pass values which need to be emitted from test. It will be more readable.